.github/workflows/ci.yml

```diff
@@ -97,30 +90,31 @@ jobs:
     - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python-version }}
-        cache: 'pip'
-        cache-dependency-path: '**/pyproject.toml'
+    - name: Install uv
+      run: pip install uv
+    - name: install python from uv
+      run: |
+        uv python install ${{ matrix.python-version }}
+        uv venv --python ${{ matrix.python-version }} .venv
     - name: Change dependencies to minimal supported versions
-      run: sed -i -e 's/>=/==/g; s/~=.*==\(.*\)/==\1/g; s/~=/==/g;' pyproject.toml
+      run: sed -i -e '/requires-python/!s/>=/==/g; /requires-python/!s/~=.*==\(.*\)/==\1/g; /requires-python/!s/~=/==/g;' pyproject.toml
       if: matrix.dependencies == 'minimal'
-    - name: Install dependencies
+    - name: Run tests
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install tox
-    # Run tox using the version of Python in `PATH`
-    - name: Run Tox with sqlite
-      run: tox -e py
-      if: matrix.database == 'sqlite'
+        python --version
+        make test
       env:
-        TESTING_SQLALCHEMY_DATABASE_URI: 'sqlite:///budget.db'
-    - name: Run Tox with postgresql
-      run: tox -e py
-      if: matrix.database == 'postgresql'
-      env:
-        TESTING_SQLALCHEMY_DATABASE_URI: 'postgresql+psycopg2://postgres:ihatemoney@localhost:5432/ihatemoney_ci'
-    - name: Run Tox with mariadb
-      run: tox -e py
-      if: matrix.database == 'mariadb'
-      env:
-        TESTING_SQLALCHEMY_DATABASE_URI: 'mysql+pymysql://root:ihatemoney@localhost:3306/ihatemoney_ci'
+        TESTING_SQLALCHEMY_DATABASE_URI: ${{ matrix.database == 'sqlite' && 'sqlite:///budget.db' || matrix.database == 'postgresql' && 'postgresql+psycopg2://postgres:ihatemoney@localhost:5432/ihatemoney_ci' || 'mysql+pymysql://root:ihatemoney@localhost:3306/ihatemoney_ci' }}
```

Comment:
I don't find this very readable, what's wrong with the current approach?

Reply:
I can add a comment there. It's just a different approach, that I find better because it has less repetition. I can surely add a comment here and format it better so it's easier to parse :-)